doc/source/users/index.rst

```diff
@@ -16,6 +16,8 @@ User's Guide
     Adding real elements <adding_elements>
     Macro Hooks <macro_hooks>
     Macro Environment <macro_environment>
+    Macro Logging <macro_logging>
```

Comment:
I have no strong opinion about this. But maybe a single chapter describing both features would be enough? At the end they are very similar.